services/workers/src/test/java/org/sagebionetworks/search/workers/sqs/search/SearchIndexWorkerIntegrationTest.java

```diff
@@ -1,67 +1,122 @@
 import org.springframework.test.context.junit.jupiter.SpringExtension;
 
+import java.util.Arrays;
+import java.util.Collections;
 import java.util.UUID;
 
+import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 
 @ExtendWith(SpringExtension.class)
 @ContextConfiguration(locations = { "classpath:test-context.xml" })
 public class SearchIndexWorkerIntegrationTest {
     private static final long MAX_WAIT = 2 * 60*1000; // 2 minutes
     private static final long CHECK_TIME = 2000;
 
-    private UserInfo adminUser;
-
     @Autowired
     private EntityManager entityManager;
     @Autowired
     private UserManager userManager;
     @Autowired
     private SearchManager searchManager;
+    @Autowired
+    private EntityAclManager entityAclManager;
+
+    private Project project;
+    private UserInfo adminUser;
+    private UserInfo anotherUser;
+
 
     @BeforeEach
     public void before() {
         adminUser = userManager.getUserInfo(AuthorizationConstants.BOOTSTRAP_PRINCIPAL.THE_ADMIN_USER.getPrincipalId());
-    }
+        String userName = UUID.randomUUID().toString();
+        anotherUser = userManager.createOrGetTestUser(adminUser, new NewUser().setUserName(userName).setEmail(userName + "@foo.org"));
 
-    @Test
-    public void testSearchIndexWorker() throws Exception {
-        Project project = new Project();
-        project.setName("OpenSearch.Project" + UUID.randomUUID());
+        project = new Project();
+        project.setName("OpenSearch Project " + UUID.randomUUID());
 
         //creating entity, trigger create message. SearchIndexWorker will pick up the message and create a searchable document.
         String id = entityManager.createEntity(adminUser, project, null);
         project = entityManager.getEntity(adminUser, id, Project.class);
-        assertNotNull(project);
+    }
+
+    @AfterEach
+    public void after(){
+        if (project != null){
+            entityManager.deleteEntity(adminUser, project.getId());
+        }
+
+        if (anotherUser != null) {
+            userManager.deletePrincipal(adminUser, anotherUser.getId());
+        }
+    }
+
+    @Test
+    public void testSearchIndexWorker() throws Exception {
 
         //call under test
-        waitForQuery(id, project.getEtag());
+        waitForEntityToAppearInSearch(project.getId(), project.getEtag());
+
+        // The admin should find the project
+        SearchResults results = searchManager.search(adminUser, searchQueryByTerm(project.getName()));
+        assertNotNull(results);
+        assertTrue(results.getHits().stream().anyMatch(hit -> hit.getId().equals(project.getId())));
 
+        // No results for the user since the project is not shared
+        SearchResults results1 = searchManager.search(anotherUser, searchQueryByTerm(project.getName()));
```

Comment:
The other should be able to see a sub-set of the project (but not all).  Zero results could mean anything.